PolyEngine/Core/Src/SafePtr.hpp

```diff
@@ -5,64 +5,65 @@
 
 namespace Poly
 {
+	/// <summary>Pointer wrapper, which returns nullptr if target object is not existing anymore</summary>
 	template<typename T>
 	class SafePtr : public BaseObject<>
 	{
 	public:
+		/// <summary>Basic constructor</summary>
+		/// <param name="pointer">Raw pointer to target object</param>
 		SafePtr(SafePtrRoot* pointer)
 		{
 			PointerIdx = SafePtrRoot::RegisterPointer(pointer);
 		}
 
+		/// <summary>Returns raw pointer to target object or nullptr, if target object do not exist</summary>
+		/// <returns>Raw pointer to target object</returns>
 		T *Get() const
 		{
 			return dynamic_cast<T*>(SafePtrRoot::GetPointer(PointerIdx));
 		}
 
-		SafePtr operator=(const SafePtrRoot* pointer)
-		{
-			return SafePtr(pointer);
-		}
+		/// <summary>Alternative way to construct SafePtr instance</summary>
+		/// <param name="pointer">Raw pointer to target object</param>
+		/// <returns>SafePtr instance</returns>
+		SafePtr operator=(const SafePtrRoot* pointer) {	return SafePtr(pointer); }
 
-		T *operator->()
-		{
-			return Get();
-		}
+		/// <summary>Member access operator</summary>
+		/// <returns>Raw pointer to target object</returns>
+		T *operator->() { return Get();	}
 
-		const T *operator->() const
-		{
-			return Get();
-		}
+		/// <summary>Constant member access operator</summary>
+		/// <returns>Constant raw pointer to target object</returns>
+		const T *operator->() const	{ return Get();	}
 
-		T *operator*()
-		{
-			return Get();
-		}
+		/// <summary>Dereference operator</summary>
+		/// <returns>Raw pointer to target object</returns>
+		T *operator*() { return Get(); }
 
-		const T *operator*() const
-		{
-			return Get();
-		}
+		/// <summary>Constant dereference operator</summary>
+		/// <returns>Constant raw pointer to target object</returns>
+		const T *operator*() const { return Get(); }
 
-		bool operator==(const SafePtr<T>& other) const 
-		{
-			return this->Get() == other.Get();
-		}
+		/// <summary>Equality operator</summary>
+		/// <param name="other">Another SafePtr instance</param>
+		/// <returns>True, if both SafePtr instances points to the same object. False, if not</returns>
+		bool operator==(const SafePtr<T>& other) const { return this->Get() == other.Get();	}
 
-		bool operator==(const T& other) const
-		{
-			return this->Get() == &other;
-		}
+		/// <summary>Equality operator</summary>
+		/// <param name="other">Raw pointer</param>
+		/// <returns>True, if both SafePtr instance and raw pointer points to the same object. False, if not</returns>
+		bool operator==(const T& other) const {	return this->Get() == &other; }
```

Comment:
This should take a  pointer to `T` not its reference, like so:
`bool operator==(const T* other) const { return this->Get() == other; }`
Same for neq operator.